libs/vm/src/analyser.cpp

```diff
@@ -1389,12 +1513,19 @@ bool Analyser::AnnotateIndexOp(ExpressionNodePtr const &node)
   }
   else
   {
-    if (AnnotateExpression(lhs) == false)
+    if (!AnnotateExpression(lhs))
     {
       return false;
     }
   }
 
+  if (lhs->node_kind == NodeKind::InitializerList)
+  {
+    // TODO(bipll): while indexing into initialization trees makes perfect easily defined sense
+    // a correct implementation would take too much for now
+    AddError(lhs->line, "indexing into initializer lists not supported yet");
+  }
+
```

Comment:
{1, 2, 3}[index] will never be an allowed expression because the type that is being indexed cannot be inferred. This if block can be removed.

Reply:
Indexing can be the initlist's own operation, without need to infer any type at all, can't it? In this example it would be quite idiomatic to allow `{1, 2, 3}[1] == 2`. 'Never' is too long a term, IMO.

Reply:
No, init lists, like null, do not have a type at all, they must always be inferred to a real type before being put to use.

Reply:
> No, init lists, like null, do not have a type at all, they must always be inferred to a real type before being put to use.

This is not an explanation or reasoning, but only a confirmation of status quo.

Reply:
OK, fair play. Never say never -- maybe in the future we can do something with this. Can we leave this if check in then, but simply remove the "yet" from the error message? That way we don't raise an expectation that it will be done soon, but don't close the possibility of doing so.

Reply:
Agreed. :-)